### PR TITLE
Improve posix module

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Add docstrings and some helper functions.

### CODE REVIEW

File: lib/posix/posix.nim [hunk 1]
```diff
@@ -592,14 +649,43 @@ proc S_TYPEISTMO*(buf: var Stat): bool {.importc, header: "<sys/stat.h>".}
...
 proc munlock*(a1: pointer, a2: int): cint {.importc, header: "<sys/mman.h>".}
+
+proc memoryUnlock*(a1: pointer, a2: int) =
+  ## Unlock pages starting from a1 for a1 bytes and allow them to be swapped.
+  if munlock(a1, a2) != 0:
+    raise newException(OSError, $strerror(errno))
+
 proc munlockall*(): cint {.importc, header: "<sys/mman.h>".}
+
+proc memoryUnockAll*() =
```
Comment (juancarlospaco) [inline lib/posix/posix.nim]: Typo `memoryUnockAll`

**Comment by Araq** [comment]
I appreciate the additions but posix.nim is supposed to only be a wrapper, not provide convenience procs. For example, with your additions it is not clear anymore that none of these procs raise exceptions, because the new ones now do raise. Please move them to a new module ``posix_utils.nim`` or similar.

File: lib/posix/posix.nim [hunk 2]
```diff
@@ -686,6 +772,13 @@ proc getrusage*(who: cint, rusage: ptr Rusage): cint
 proc bsd_signal*(a1: cint, a2: proc (x: pointer) {.noconv.}) {.
   importc, header: "<signal.h>".}
 proc kill*(a1: Pid, a2: cint): cint {.importc, header: "<signal.h>".}
+
+proc kill*(pid: Pid, signal: int) =
+  ## Sends a signal to a running process.
+  ## Raise exception in case of failure e.g. process not running.
+  if kill(pid, signal.cint) != 0:
+    raise newException(OSError, $strerror(errno))
+
```
Comment (krux02) [inline lib/posix/posix.nim]: This overload is a breaking change. Programs that have been written to test the return value won't compile anymore.

**Comment by juancarlospaco** [comment]
If the code is homeless and is not getting into std lib, 
I maintain `psutil` module that has a posix module extends std libs one, more than welcome to add it there. :slightly_smiling_face: Awesome work whatsoever.

File: lib/posix/posix_utils.nim [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  ## Sends a signal to a running process by calling `kill`.
+  ## Raise exception in case of failure e.g. process not running.
+  if kill(pid, signal.cint) != 0:
+    raise newException(OSError, $strerror(errno))
+
+proc mkstemp*(prefix: string): (string, File) =
+  ## Creates a unique temporary file from a prefix string. Adds a six chars suffix.
+  ## The file is created with perms 0600.
+  ## Returs the filename and a file opened in r/w mode.
+  var tmpl = cstring(prefix & "XXXXXX")
```
Comment (juancarlospaco) [inline lib/posix/posix_utils.nim]: Why that hardcoded `"XXXXXX"` ?. :thinking:

Comment (FedericoCeratto) [reply lib/posix/posix_utils.nim]: It's a simple template format. The 6 X chars are then replaced with a randomly generated string by the OS while ensuring that the new file or dir are not clashing with existing ones.

Comment (juancarlospaco) [reply lib/posix/posix_utils.nim]: Ok, as long as it works Ok if I pass `"XXXXXX"` as `prefix` :smile:

Comment (FedericoCeratto) [reply lib/posix/posix_utils.nim]: The kernel would still parse and replace the last 6 'X' leading to a name like XXXXXXcdKdWF, which is to be expected.

**Comment by Araq** [comment]
Looks good to me. Please also add it to ``kochdocs.nim`` and the changelog.

**Comment by FedericoCeratto** [comment]
@Araq updated. Should the entries in tools/kochdocs.nim sorted alphanumerically later on?

**Comment by Araq** [comment]
> @Araq updated. Should the entries in tools/kochdocs.nim sorted alphanumerically later on?

Nah.

**Pull request merged**